text/0000-clippy-uno.md

```diff
@@ -0,0 +1,263 @@
+
+I don't think the compler will want _all_ correctness lints here, however if the lint is about a common enough situation
+where it being _not_ a bug is an exceedingly rare case (i.e. very low false positive frequency) it should probably belong in the
+compiler.
+
+ [cat]: #lint-categorization
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+
+## Lint categorization
+
+This categorization can be browsed [online].
+
+ [online]: http://rust-lang-nursery.github.io/rust-clippy/current/
+
+Please leave comments on thoughts about these lints -- if their categorization is correct, if they should exist at all, and if we should be uplifting them to the compiler.
+
+
+### correctness (Deny)
+* [naive_bytecount](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#naive_bytecount) (use of naive `<slice>.filter(|&x| x == y).count()` to count byte values)
+* [map_entry](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#map_entry) (use of `contains_key` followed by `insert` on a `HashMap` or `BTreeMap`)
+* [boxed_local](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#boxed_local) (using `Box<T>` where unnecessary)
+* [large_enum_variant](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#large_enum_variant) (large size difference between variants on an enum)
+* [manual_memcpy](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#manual_memcpy) (manually copying items between slices)
+* [unused_collect](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#unused_collect) (`collect()`ing an iterator without using the result; this is usually better written as a for loop)
+* [expect_fun_call](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#expect_fun_call) (using any `expect` method with a function call)
+* [iter_nth](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#iter_nth) (using `.iter().nth()` on a standard library type with O(1) element access)
+* [or_fun_call](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#or_fun_call) (using any `*or` method with a function call, which suggests `*or_else`)
+* [single_char_pattern](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#single_char_pattern) (using a single-character str where a char could be used, e.g. `_.split("x")`)
+* [cmp_owned](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#cmp_owned) (creating owned instances for comparing with others, e.g. `x == "foo".to_string()`)
+* [mutex_atomic](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#mutex_atomic) (using a mutex where an atomic value could be used instead)
+* [box_vec](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#box_vec) (usage of `Box<Vec<T>>`, vector elements are already on the heap)
+* [useless_vec](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#useless_vec) (useless `vec!`)
```

Comment:
I don't see [`temporary_cstring_as_ptr`](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#temporary_cstring_as_ptr) which is definitely something I would like uplifted.
IIRC at the time of the creation of this lint we had found a quite few instances of this, for which we opened PRs, but new instances might have appeared by now.

Reply:
It's not generally wrong if used as a function argument, it's "just" a very common source of dangling pointers because a simple refactoring can easily turn correct code into UB. Any use of `CString::new(...).unwrap().as_ptr()` outside a function argument is definitely wrong.

More detailed discussion can be found [here](https://github.com/rust-lang/rfcs/pull/1642)